src/Uno.UI.RemoteControl.Host/RemoteControlServer.cs

```diff
@@ -12,55 +12,103 @@
 using Uno.UI.RemoteControl.Messages;
 using System.Runtime.Loader;
 using Microsoft.Extensions.Configuration;
+using System.Reflection;
+using System.Runtime.Remoting.Messaging;
 
 namespace Uno.UI.RemoteControl.Host
 {
 	internal class RemoteControlServer : IRemoteControlServer, IDisposable
 	{
-		private readonly Dictionary<string, IServerProcessor> _processors = new Dictionary<string, IServerProcessor>();
+		private readonly Dictionary<string, IServerProcessor> _processors = new();
 
 		private System.Reflection.Assembly? _loopAssembly;
 		private WebSocket? _socket;
+		private string? _appInstanceId;
 		private readonly IConfiguration _configuration;
-		private readonly AssemblyLoadContext _loadContext;
+		private readonly static Dictionary<string, AssemblyLoadContext> _loadContexts = new();
 
 		public RemoteControlServer(IConfiguration configuration)
 		{
 			_configuration = configuration;
-			_loadContext = new AssemblyLoadContext(null, isCollectible: true);
-			_loadContext.Unloading += (e) =>
+
+			if (this.Log().IsEnabled(LogLevel.Debug))
+			{
+				this.Log().LogDebug("Starting RemoteControlServer");
+			}
+		}
+
+		string IRemoteControlServer.GetServerConfiguration(string key)
+			=> _configuration[key] ?? "";
+
+		private void EnsureLoadContextForApplicationId(string applicationId)
+		{
+			if (_loadContexts.ContainsKey(applicationId))
+			{
+				return;
+			}
+
+			var loadContext = new AssemblyLoadContext(null, isCollectible: true);
+			loadContext.Unloading += (e) =>
 			{
 				if (this.Log().IsEnabled(LogLevel.Debug))
 				{
 					this.Log().LogDebug("Unloading assembly context");
 				}
 			};
-			_loadContext.Resolving += (context, assemblyName) =>
+
+			// Add custom resolving so we can find dependencies even when the processor assembly
+			// is built for a different .net version than the host process.
+			loadContext.Resolving += (context, assemblyName) =>
 			{
 				if (_loopAssembly is not null)
 				{
-					return context.LoadFromAssemblyPath(Path.Combine(Path.GetDirectoryName(_loopAssembly.Location)!, assemblyName.Name + ".dll"));
-				}
-				else
-				{
-					return context.LoadFromAssemblyName(assemblyName);
+					try
+					{
+						var loc = _loopAssembly.Location;
+						if (!string.IsNullOrWhiteSpace(loc))
+						{
+							var dir = Path.GetDirectoryName(loc);
+							if (!string.IsNullOrEmpty(dir))
+							{
+								var relPath = Path.Combine(dir, assemblyName.Name + ".dll");
+								if (File.Exists(relPath))
+								{
+									return context.LoadFromAssemblyPath(relPath);
```

Comment:
It would be worth adding a Trace in here to log out the path to the assembly that's been loaded